bsp/esp_bsp_generic/Kconfig

```diff
@@ -479,7 +518,283 @@ menu "Board Support Package"
+                help
+                    The CS pin for SPI display.
+            
+            config BSP_DISPLAY_DC_GPIO
+                depends on BSP_DISPLAY_INTERFACE_SPI
+                int 
+                prompt "Display SPI DC GPIO"
+                default 0
+                range -1 ENV_GPIO_OUT_RANGE_MAX
+                help
+                    The DC pin for SPI display.
+            
+            config BSP_DISPLAY_RST_GPIO
+                int 
+                prompt "Display reset GPIO"
+                default 0
+                range -1 ENV_GPIO_OUT_RANGE_MAX
+                help
+                    The reset pin for display.
+            
+            config BSP_DISPLAY_BACKLIGHT_GPIO
+                int 
+                prompt "Display backlight GPIO"
+                default 0
+                range -1 ENV_GPIO_OUT_RANGE_MAX
+                help
+                    The backlight pin for display.
+            
+        endmenu
+            
+        choice  
+            prompt "Display driver"
+            depends on BSP_DISPLAY_ENABLED
+            help
+                Select a display driver 
+            config BSP_DISPLAY_DRIVER_ST7789
+                depends on BSP_DISPLAY_INTERFACE_SPI
+                bool "ST7789"
+            config BSP_DISPLAY_DRIVER_ILI9341
+                depends on BSP_DISPLAY_INTERFACE_SPI
+                bool "ILI9341"
+            config BSP_DISPLAY_DRIVER_GC9A01
+                depends on BSP_DISPLAY_INTERFACE_SPI
+                bool "GC9A01"
+        endchoice
+        
+        menu "Rotation"
+            depends on BSP_DISPLAY_ENABLED
+
+            config BSP_DISPLAY_ROTATION_SWAP_XY
+                bool "Swap XY"
+                default n
+                help
+                    Swap X and Y axis on the screen.
+
+            config BSP_DISPLAY_ROTATION_MIRROR_X
+                bool "Mirror X"
+                default n
+                help
+                    Mirror X axis on the screen.
+
+            config BSP_DISPLAY_ROTATION_MIRROR_Y
+                bool "Mirror Y"
+                default n
+                help
+                    Mirror Y axis on the screen.
+        endmenu
+            
+        choice  
+            prompt "Color space"
+            depends on BSP_DISPLAY_ENABLED
+            help
+                Select a display color space 
+            config BSP_DISPLAY_COLOR_SPACE_RGB
+                bool "RGB"
+            config BSP_DISPLAY_COLOR_SPACE_BGR
+                bool "BGR"
+        endchoice
+
+        config BSP_DISPLAY_CMD_BITS
+            depends on BSP_DISPLAY_ENABLED
+            int "LCD command bits"
+            default 8
+            help
+                The command bits of the display.
+
+        config BSP_DISPLAY_PARAM_BITS
+            depends on BSP_DISPLAY_ENABLED
+            int "LCD param bits"
+            default 8
+            help
+                The param bits of the display.
```

Comment:
This should be constant for given LCD controller. Eg. for ILI9341 this is always 8, right?

Reply:
Sorry, I did not write it clearly. Could we hide this config option?

Reply:
Sure. Done.